src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.iOS.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.IO
+{
+    public static partial class Path
+    {
+        private static string s_defaultTempPath = string.Empty;
+
+        private static string DefaultTempPath =>
+            s_defaultTempPath ?? (s_defaultTempPath = Interop.Sys.SearchPathTempDirectory()) ??
+            ThrowHelper.ThrowInvalidOperationException();
```

Comment:
This can't be valid, surely? `ThrowHelper.ThrowInvalidOperationException()` returns void, it's not an appropriate thing to use `??` with

Reply:
https://github.com/dotnet/csharplang/blob/main/meetings/2018/LDM-2018-07-16.md#throw-expression-on-the-right-hand-side

Reply:
It surely is, this syntax should work on everything (since netstandard2.1?).. [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgAIqKBYAKHoGI8cAbVgQ2FZgpjy4969AQFsYAZwAOHMLwCSeDDCgQp9AN71qFAJZKVA1hVIUAygE8J26lro6d+5VCMmySNwAZzMDlDAAFgAKHBgBACowolIAIrqwYBjQFgAUAJQUALwAfBQARAD0AGYQEHkA3DZUAL70tQx0RADMbh4yUBi6HMamIWGaVRRSULoAbqG8RO5eAPwUEgD6ACYwRRxsGJHRfQGVDY4GLt2tXhQxq+usm1FSO1nZgzqLK2sbW7ehARQzcynPF28bndMhRFM41AA6SwSKG+fzBT7vOIJJJQVJpDI/R7UMKqADuFDwMAJinGrF0SwA8lIVKFdBA8ABRBByKSdBnpPb1eimABiA3s1CmHiIKAoAFkOPp0vc3ABOFI7CHnV5Xd47NJcoA). Does it not work with version of csc being used with mono? is it a different version?

Reply:
ah, just use `throw new InvalidatiOperationException()` here. No need for helper since it's not a hot path (when exception is being thrown in this case). Also, `string? s_defaultTempPath = null` is correct. string.Empty is wrong.